subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy

```diff
@@ -82,7 +105,16 @@ class BaselineVersion implements VersionResults {
     }
 
     private static boolean differenceIsSignificant(DataSeries<Duration> myTime, DataSeries<Duration> otherTime, double minConfidence) {
-        DataSeries.confidenceInDifference(myTime, otherTime) > minConfidence
+        return (myTime.median - otherTime.median).abs() > MINIMUM_DIFFERENCE_WE_CAN_MEASURE &&
+            (relativeDifferenceInMedianIsVeryHigh(myTime, otherTime) || DataSeries.confidenceInDifference(myTime, otherTime) > minConfidence)
     }
 
+    private static boolean differenceInMedianIsSignificant(DataSeries<Duration> myTime, DataSeries<Duration> otherTime, double minRelativeMedianDifference) {
+        def minimumMedian = [myTime.median, otherTime.median].min()
+        return ((myTime.median - otherTime.median) / minimumMedian).abs() > minRelativeMedianDifference
```

Comment:
does this handle also zero or some such corner values?

Reply:
It does not. Do you think handling of 0 is necessary? That would mean we have a Gradle invocation which takes 0 milliseconds every time.

Reply:
I was thinking if maybe measurement code would report 0, due to some bug, or some other weird case